confluent_kafka/avro/serializer/message_serializer.py

```diff
@@ -216,12 +216,18 @@ def decode_message(self, message, is_key=False):
         if message is None:
             return None
 
-        if len(message) <= 5:
+        if len(message) <= 5 and not is_key:
             raise SerializerError("message is too small to decode")
 
         with ContextStringIO(message) as payload:
             magic, schema_id = struct.unpack('>bI', payload.read(5))
             if magic != MAGIC_BYTE:
-                raise SerializerError("message does not start with magic byte")
+                if not is_key:
+                    raise SerializerError("message does not start with magic byte")
+                key_str = payload.getvalue().decode('utf-8')
+                print('Warning: decoding key as string', key_str)
```

Comment:
Thoughts on `log.warning("decoding key as string: {key_str}")` (or whatever string formatting is most appropriate)?

Reply:
I don't think it is necessary to even log, it should just be documented and be the expected behavior, unless there is a case where it could break being decoded as string.